# x and y are coordinates in the grid starting from 0
float32 x
float32 y

float32 thetaQuaternion

float32 theta

# orientation should be one of the following constants
int32 orientation

int32 RIGHT = 0
int32 UP = 1
int32 LEFT = 2
int32 DOWN = 3
